src/functions/Pester.SessionState.Mock.ps1

```diff
@@ -219,15 +223,22 @@ function Mock {
     .LINK
     https://pester.dev/docs/usage/mocking
     #>
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'Default')]
+    # TODO: Breaking change due to parameter sets. Previously had positions:
+    # CommandName, MockWith, ParameterFilter, ModuleName, RemoveParameterType, RemoveParameterValidation
     param(
+        [Parameter(Position = 0)]
         [string]$CommandName,
+        [Parameter(Position = 1)]
         [ScriptBlock]$MockWith = {},
         [switch]$Verifiable,
+        [Parameter(ParameterSetName = 'WithParameterFilter')]
         [ScriptBlock]$ParameterFilter,
         [string]$ModuleName,
         [string[]]$RemoveParameterType,
-        [string[]]$RemoveParameterValidation
+        [string[]]$RemoveParameterValidation,
+        [Parameter(ParameterSetName = 'WithParameterFilter')]
+        [switch]$AllowFallback
```

Comment:
Do we want to support all original positional parameters or are `CommandName` + `MockWith` enough? Wouldn't expect the rest defined at line 228 to be commonly used.